generators/pythonv2/codegen/src/ast/core/Writer.ts

```diff
@@ -14,6 +15,11 @@ export class Writer extends AbstractWriter {
         return this.buffer;
     }
 
+    public async toStringFormatted(): Promise<string> {
+        await init();
+        return format(this.buffer);
+    }
```

Comment:
I intentionally chose to add this to the writer rather than the AstNode. This is because the writer has context needed for codegen that the AstNode doesn't have – primarily, references.

This was important for `PythonFile`, which relies on the fact that the writer has accumulated the references needed to generate imports for the file.

Reply:
Side Note: I wonder if there's a way to use a global to ensure that in a given process, we only ever call `await init();` once? I noticed tests were a bit slower and I suspect this is why.

Reply:
```
export abstract class AstNode extends AbstractAstNode {
    /**
     * Writes the node to a string.
     */
    public toString(): string {
        const writer = new Writer();
        this.write(writer);
        return writer.toString();
    }

    public toStringFormatted(): string {
        return prettier.format(this.toString(), { parser: "typescript", tabWidth: 4, printWidth: 120 });
    }
}
```

Reply:
this is how i do it in typescript